### PR TITLE
raised resource class of `t_ubu_ubsan_clang_cli` to medium

### LINKED ISSUE #13252: solfuzzer crashes in nightly
Our nightly test `t_ubu_ubsan_clang_cli` [fails sometimes](https://app.circleci.com/pipelines/github/ethereum/solidity/25120/workflows/e9f4961f-a6aa-4071-8123-22a76515d86b/jobs/1106669)  with 

```
Testing soljson via the fuzzer...
xargs: /root/project/build/test/tools/solfuzzer: terminated by signal 9
```

### PR DESCRIPTION
**Pull request by nishant-sachdeva**
closes #13252

### CODE REVIEW

File: .circleci/config.yml
Comment (nikola-matic) [inline .circleci/config.yml]: From what I can tell, this step inherits from `base_ubuntu2004_clang_small` and then overrides the resource class with `medium`. `base_ubuntu2004_clang_small` is however only used once in this file, so it might make sense to rename it to `base_ubuntu2004_clang_medium` and set the resource class to `medium` in it directly.
It would be nice if @r0qs or @cameel were to take a look, since I'm far from a devops/circleci expert.

Comment (cameel) [reply .circleci/config.yml]: Exactly. You should define `base_ubuntu2004_clang_medium` instead of adding the resource class here.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `dc7ee5ceb79e27254bebd9de4aa1bf7ab13a95f2`:
```
Error: Trailing whitespace found:
.circleci/config.yml:321:  

```
Please check that your changes are working as intended.

File: .circleci/config.yml
Comment (nishant-sachdeva) [reply .circleci/config.yml]: Added these changes in the latest push. 

Should we go ahead and increase `MAKEFLAGS: -j 2` to a higher number ?  I noticed that compared the resource class `large` seems to have a higher allowance for the `-j` flag.

File: .circleci/config.yml
Comment (nikola-matic) [inline .circleci/config.yml]: You can delete this and just rename the `small` resource class to `medium`, and set its `resource_class` to `medium`, since it isn't used anywhere anymore.

File: .circleci/config.yml
Comment (nikola-matic) [reply .circleci/config.yml]: Looks like `small` resource classes usually have `-j2` set, and `large` have `-j5`, so I'd go with `-j3` here - it's default for the base `base_ubuntu2004_clang` anyway, so technically you don't even need to specify it (even though it would be nice for readability).

Comment (nishant-sachdeva) [reply .circleci/config.yml]: Ah, yes. Just noticed that the entire `environment` section is redundant. Will remove all of it and let it set itself to default values.

File: .circleci/config.yml
Comment (nikola-matic) [inline .circleci/config.yml:316]: I am conflicted. This is exactly what you should do (deleting redundant/inherited fields) even though it impacts readability, but we seem to be using the redundancy approach all over this file when defining containers. So if readability is of paramount importance, then we might as well not use a base definition of the container/step/whatever if we're gonna overwrite all of the fields with the same values again.

**Review by nikola-matic — DISMISSED** [review]
I am personally fine with this approach, which is why I'm approving, but @cameel may disagree, so please wait for his final review. My 'concerns' are in [this comment](https://github.com/ethereum/solidity/pull/13655#discussion_r1006514626).

File: .circleci/config.yml
Comment (cameel) [inline .circleci/config.yml:319]: It's no longer really `base_ubuntu2004_`**clang**`_medium` if you remove the clang part :P

File: .circleci/config.yml
Comment (cameel) [reply .circleci/config.yml:316]: @nishant-sachdeva Please restore the environment section. Providing values for these variables is part of the point of having these base definitions here.

File: .circleci/config.yml
Comment (cameel) [inline .circleci/config.yml]: I just realized that we already have a definition for `medium`. It's `base_ubuntu2004_clang` above (`medium` is the default). So you actually don't have to add a new definition. Just use that one.

**Review by cameel — CHANGES_REQUESTED** [review]

File: .circleci/config.yml
Comment (nishant-sachdeva) [reply .circleci/config.yml]: That sounds better. Will remove the definition entirely in the latest push.

**Review by cameel — APPROVED** [review]

**Pull request merged**
